optimum/onnxruntime/modeling_ort.py

```diff
@@ -193,11 +204,12 @@ def _from_pretrained(
                 kwargs will be passed to the model during initialization
         """
         config_dict = kwargs.pop("config", {})
+        provider = kwargs.pop("provider", None)
         model_file_name = file_name if file_name is not None else ONNX_WEIGHTS_NAME
         # load model from local directory
         if os.path.isdir(model_id):
             config = PretrainedConfig.from_dict(config_dict)
-            model = ORTModel.load_model(os.path.join(model_id, model_file_name))
+            model = ORTModel.load_model(os.path.join(model_id, model_file_name), provider=provider)
```

Comment:
We should also make sure that when a user provides a provider `CUDA` or `TensorRT` that the device will be `cuda`. Meaning that the model is "moved" automatically/the device is set correctly